### PR TITLE
SOLR-16611: fix collapse hint=top_fc bug

### PR DESCRIPTION
**Pull request by metapox**
# Description

https://issues.apache.org/jira/browse/SOLR-16611 

# Solution

Changed uninvertingReader#getSorted so that it does not return null.  

# Tests/

Added unit test for no docs.
The test now succeeds, but the code before the change fails.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.

### CODE REVIEW

**Review by mkhludnev — COMMENTED** [review]
I think it's worth to add a test case.

File: solr/core/src/test/org/apache/solr/search/TestCollapseQParserPlugin.java [hunk 1]
```diff
@@ -440,6 +440,13 @@ private void testDoubleCollapse(String group) {
   }
 
   private void testCollapseQueries(String group, String hint, boolean numeric) {
+    // Test no docs  https://issues.apache.org/jira/browse/SOLR-16611
```
Comment (mkhludnev) [inline solr/core/src/test/org/apache/solr/search/TestCollapseQParserPlugin.java:443]: pls remove comment before commit

**Review by mkhludnev — APPROVED** [review]

**Comment by mkhludnev** [comment]
@metapox can you confirm that you run tests? Pls put to comment the failed ones. Just to double check before merge. 
Appreciate your effort!

**Comment by metapox** [comment]
@mkhludnev Thanks for review!

> Pls put to comment the failed ones.

Are you satisfied with this?

before
```
./gradlew test --tests org.apache.solr.search.TestCollapseQParserPlugin

> Task :randomizationInfo
Running tests with randomization seed: tests.seed=1695D71ECE38B312

> Task :errorProneSkipped
WARNING: errorprone disabled (skipped on builds not running inside CI environments, pass -Pvalidation.errorprone=true to enable)

> Task :solr:core:test
WARNING: A command line option has enabled the Security Manager
WARNING: The Security Manager is deprecated and will be removed in a future release

org.apache.solr.search.TestCollapseQParserPlugin > testStringCollapse FAILED
    java.lang.RuntimeException: Exception during query
        at __randomizedtesting.SeedInfo.seed([1695D71ECE38B312:E31DE77F075F1006]:0)
        at org.apache.solr.SolrTestCaseJ4.assertQ(SolrTestCaseJ4.java:1044)
        at org.apache.solr.SolrTestCaseJ4.assertQ(SolrTestCaseJ4.java:1002)
        at org.apache.solr.search.TestCollapseQParserPlugin.testCollapseQueries(TestCollapseQParserPlugin.java:446)
        at org.apache.solr.search.TestCollapseQParserPlugin.testStringCollapse(TestCollapseQParserPlugin.java:224)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:577)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.SystemPropertiesRestoreRule$1.evaluate(SystemPropertiesRestoreRule.java:80)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.SystemPropertiesRestoreRule$1.evaluate(SystemPropertiesRestoreRule.java:80)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base/java.lang.Thread.run(Thread.java:833)

        Caused by:
        java.lang.NullPointerException: Cannot invoke "org.apache.lucene.index.SortedDocValues.getValueCount()" because "this.collapseValues" is null
            at org.apache.solr.search.CollapsingQParserPlugin$OrdScoreCollector.<init>(CollapsingQParserPlugin.java:635)
            at org.apache.solr.search.CollapsingQParserPlugin$CollectorFactory.getCollector(CollapsingQParserPlugin.java:2145)
            at org.apache.solr.search.CollapsingQParserPlugin$CollapsingPostFilter.getFilterCollector(CollapsingQParserPlugin.java:469)
            at org.apache.solr.search.SolrIndexSearcher.getProcessedFilter(SolrIndexSearcher.java:1275)
            at org.apache.solr.search.SolrIndexSearcher.getDocListNC(SolrIndexSearcher.java:1793)
            at org.apache.solr.search.SolrIndexSearcher.getDocListC(SolrIndexSearcher.java:1678)
            at org.apache.solr.search.SolrIndexSearcher.search(SolrIndexSearcher.java:712)
            at org.apache.solr.handler.component.QueryComponent.doProcessUngroupedSearch(QueryComponent.java:1637)
            at org.apache.solr.handler.component.QueryComponent.process(QueryComponent.java:421)
            at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:419)
            at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:224)
            at org.apache.solr.core.SolrCore.execute(SolrCore.java:2874)
            at org.apache.solr.util.TestHarness.query(TestHarness.java:357)
            at org.apache.solr.util.TestHarness.query(TestHarness.java:337)
            at org.apache.solr.SolrTestCaseJ4.assertQ(SolrTestCaseJ4.java:1016)
            ... 47 more

....



> Task :solr:core:test FAILED

ERROR: The following test(s) have failed:
  - org.apache.solr.search.TestCollapseQParserPlugin.testStringCollapse (:solr:core)
    Test output:/path/solr/solr/core/build/test-results/test/outputs/OUTPUT-org.apache.solr.search.TestCollapseQParserPlugin.txt
    Reproduce with: gradlew :solr:core:test --tests "org.apache.solr.search.TestCollapseQParserPlugin.testStringCollapse" -Ptests.jvms=8 "-Ptests.jvmargs=-XX:TieredStopAtLevel=1 -XX:+UseParallelGC -XX:ActiveProcessorCount=1 -XX:ReservedCodeCacheSize=120m" -Ptests.seed=1695D71ECE38B312 -Ptests.file.encoding=ISO-8859-1


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':solr:core:test'.
> There were failing tests. See the results at: /path/solr/solr/core/build/test-results/test/

```




now
```
./gradlew test --tests org.apache.solr.search.TestCollapseQParserPlugin
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :randomizationInfo
Running tests with randomization seed: tests.seed=BF220AB80572D5AE

> Task :errorProneSkipped
WARNING: errorprone disabled (skipped on builds not running inside CI environments, pass -Pvalidation.errorprone=true to enable)

> Task :solr:core:test
WARNING: A command line option has enabled the Security Manager
WARNING: The Security Manager is deprecated and will be removed in a future release
:solr:core:test (SUCCESS): 15 test(s)
The slowest tests (exceeding 500 ms) during this run:
   1.02s TestCollapseQParserPlugin.testNullGroupNumericVsStringCollapse (:solr:core)
   0.77s TestCollapseQParserPlugin.testStringCollapse (:solr:core)
   0.60s TestCollapseQParserPlugin.testNumericCollapse (:solr:core)
The slowest suites (exceeding 1s) during this run:
   7.49s TestCollapseQParserPlugin (:solr:core)

BUILD SUCCESSFUL in 32s
159 actionable tasks: 65 executed, 94 up-to-date
```

**Comment by mkhludnev** [comment]
Thanks, @metapox. Your test is great. 
The issue is sometimes a change causes failures in other quite unexpected tests, and it goes to urgent fixes after a merge. So, I really appreciate if you run tests. Unusually, there are some _false negative_ failures in cluster tests, timeouts etc. But that how it works.

**Comment by mkhludnev** [comment]
@metapox , how I can attribute your contribution in CHANGES.txt? Please give your name or desired login.

**Comment by mkhludnev** [comment]
Hi, @metapox I run tests myself. Failures seem unrelated to the change. 
Please confirm I put your name correctly in CHANGES. I'll merge then.

**Comment by metapox** [comment]
@mkhludnev 
Hi, @mkhludnev 
My name you put in CHANGES  is correct.
Thank you!

**Pull request merged**